lib/controllers/article.dart

```diff
@@ -0,0 +1,58 @@
+import 'package:get/get.dart';
+import 'package:project/models/article.dart';
+
+import 'controller.dart';
+
+typedef JsonArticle = Map<String, String>;
+typedef JsonArticleList = List<JsonArticle>;
+
+class ArticleController extends AppController {
+  @override
+  String storeId = 'articles';
+
+  final List<Article> _articles = <Article>[].obs;
+  List<Article> get articles => _articles;
+
+  List<dynamic> _jsonList = <dynamic>[];
+
+  @override
+  Future<void> onInit() async {
+    await super.onInit();
+    _restore();
+  }
+
+  void _restore() {
+    _jsonList = storage;
+
+    for (final dynamic article in _jsonList) {
+      _articles.add(Article.fromJson(article));
+    }
+  }
+
+  void erase() {
+    _articles.clear();
+    _jsonList.clear();
+    storage = _jsonList;
+  }
+
+  void create(Article article) {
+    _articles.add(article);
+
+    _jsonList.add(article.toJson());
+    storage = _jsonList;
+  }
+
+  Article? find(String id) => _articles.firstWhere(
+        (Article article) => article.id == id,
+      );
+
+  Iterable<Article> search(String title) => _articles.where(
+        (Article article) => article.title == title,
```

Comment:
Isn't it better to make a ``.containt`` ? because there you have to type the exact title

Reply:
The `.contains` method takes an `Article` as parameter, but this function is made to search by title.
This is a function used inside the search_screen, where we will call this function with the user input.

Reply:
Oh no ok I see what you mean, sorry. i'll change that

Reply:
Exactly, if the title of an article is "Magnifique" and the user writes "my" we do not have the article
I propose :
```
Iterable<Article> search(String title) => _articles.where(
        (Article article) => article.title.contains(title),
);
```